### DISCUSSION TITLE
How complicated would it be to call from Golang?

### DISCUSSION THREAD

**Discussion by canselcik** [rune-rs/rune#818] [General]
I have Go project in which I use Lua to do some scriptable things. I think Rune is better to work with in every way but I don't want to have to write the whole backend in Rust for this, although it felt appealing at times, it would be quite a large undertaking.

How much work do you think it would be to call Golang from Rune and vice versa? Full interop with types.

**Comment by udoprog** [comment]
Not doable without finishing https://github.com/rune-rs/rune/pull/380

But then, about as hard as calling any other C API from Go.
